src/devices/Tca9548a/Tca9548A.cs

```diff
@@ -0,0 +1,229 @@
+            _shouldDispose = shouldDispose;
+        }
+
+        /// <summary>
+        /// Select a multiplex channel
+        /// </summary>
+        /// <param name="multiplexChannel">The channel to select [2^channel number (0-7)]</param>
+        public void SelectChannel(byte multiplexChannel)
+        {
+            try
+            {
+                _i2CDevice.WriteByte(multiplexChannel);
+                _i2CDevice.ReadByte();
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine($"Unable to select ({_i2CDevice.ConnectionSettings.BusId} ,{_i2CDevice.ConnectionSettings.DeviceAddress}, {multiplexChannel}): {e}");
+            }
+        }
+
+        /// <summary>
+        /// Select a multiplex channel
+        /// </summary>
+        /// <param name="multiplexChannel">The channel to select</param>
+        public void SelectChannel(Channels multiplexChannel)
+        {
+            SelectChannel(Convert.ToByte(multiplexChannel));
+        }
+
+        /// <summary>
+        /// Try getting the selected channel on Multiplexer
+        /// </summary>
+        /// <param name="selectedChannel"> selected Multiplexer Channel</param>
+        /// <returns> true if able to retrieve selected channel</returns>
+        public bool TryGetSelectedChannel(out Channels selectedChannel)
+        {
+            try
+            {
+                var channel = _i2CDevice.ReadByte();
+                if (Enum.IsDefined(typeof(Channels), channel))
+                {
+                    selectedChannel = (Channels)channel;
+                    return true;
+                }
+                else
+                {
+                    selectedChannel = Channels.Channel0;
+                    return false;
+                }
+            }
+            catch (Exception)
+            {
+                selectedChannel = Channels.Channel0;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Returns all the connected devices on all channels of Multiplexer
+        /// </summary>
+        /// <returns> Dictionary having Keys as Channels and Values as IEnumerable integer type</returns>
+        public Dictionary<Channels, IEnumerable<int>> ScanAllChannelsForDeviceAddress()
+        {
+            Dictionary<Channels, IEnumerable<int>> channels = new Dictionary<Channels, IEnumerable<int>>();
+            foreach (Channels channel in DeviceChannels)
+            {
+                channels.Add(channel, ScanChannelsForDeviceAddress(channel));
+            }
+
+            return channels;
+        }
+
+        /// <summary>
+        /// Returns all the connected device on selected channel
+        /// </summary>
+        /// <param name="channel">Selected Channel on Multiplexer</param>
+        /// <returns></returns>
+        public IEnumerable<int> ScanChannelsForDeviceAddress(Channels channel)
```

Comment:
You should be able to replace this with a call to `I2cBusExtensions.PerformBusScan` (in the `Iot.Device.Boards` namespace). That does the same already.

Thinking about it... Since that method takes an `I2cBus` as argument, I think it would make sense to add a method `public I2cBus CreateBusFromChannel(Channels channel)` with something like `return new I2cBus(<ArbitraryNumberToIdentifyChannel>)`

Reply:
I have added an extension function in I2cBusExtensions class to get an I2cBus object from I2cDevice object